spec/acceptance/inputs_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'spec_helper_acceptance'
+
+describe 'Rsyslog inputs' do
+  before(:context) do
+    cleanup_helper
+  end
+
+  context 'basic input' do
+    it 'applies with inputs' do
+      pp = <<-MANIFEST
+      class { 'rsyslog::server':
+        inputs => {
+          'imudp' => {
+            'type'  => 'imudp',
+            'config' => {
+              'port' => '514',
+            },
+          },
+          'imptcp' => {
+            'type'  => 'imptcp',
+            'config' => {
+              'port' => '514',
+            },
+          },
+        },
+      }
+      MANIFEST
+
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    describe file('/etc/rsyslog.d/50_rsyslog.conf') do
+      its(:content) { is_expected.to contain('input\(type="imudp"\n\s*port="514"\n\)') }
+      its(:content) { is_expected.to contain('input\(type="imptcp"\n\s*port="514"\n\)') }
+    end
+  end
+
+  context 'inputs with custom priorities' do
+    it 'applies with custom priorities' do
+      pp = <<-MANIFEST
+class { 'rsyslog::server':
+        inputs => {
+          'imfile' => {
+            'priority' => 10,
+            'type'     => 'imfile',
+            'config'   => {
+              'File' => '/tmp/test-file',
+            },
+          },
+          'imudp' => {
+            'priority' => 111,
+            'type'     => 'imudp',
+            'config'   => {
+              'port' => '514',
+            },
+          },
+          'imptcp' => {
+            'priority' => 112,
+            'type'     => 'imptcp',
+            'config'   => {
+              'port' => '514',
+            },
+          },
+          'imfile2' => {
+            'priority' => 50,
+            'type'     => 'imfile',
+            'config'   => {
+              'File' => '/tmp/test-file2',
+            },
+          },
+        },
+      }
+      MANIFEST
+
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    describe file('/etc/rsyslog.d/50_rsyslog.conf') do
+      its(:content) { is_expected.to contain('input\(type="imfile"\n\s*File="/tmp/test-file"\n\)\n# imfile2\n') }
```

Comment:
I'm worried this isn't actually testing the proper order of the various fragments.

Reply:
It's testing based on the next comment in the file. The comment is always the first part of the next config option. There's no need to test all the way down if there is a consistent structure in the templates and that structure is being tested elsewhere

Reply:
It's been a long day - you're right that's a valid way of testing.